### PR TITLE
Feature/3016 reporting dates table template

### LINKED ISSUE #3016: Design a new table in Wagtail to display the 2020 congressional pre-primary reports info
### Summary

_Looking to create a table in Wagtail that is user friendly and easy to update the 2020 reporting information. The table will be used on the child pages to display reporting information under Dates and deadlines._

### Proposed Behavior / Feature

- [x] Looking to create a dropdown for an individual to pick a state to view reporting information.
- [x] Looking for an option to display all 50 states and territories, so the regulated community can also see the big picture. 
- [x] Decrease possibility of human error by creating a table that is easy to input data and display information clearly using Wagtail.
- [x] Choose tool we'll use to populate data.

### Current Situation

_Current situation is we are using straight html code in the template. The current table for pre-election report information is huge. The increased ability for human error is a factor, and it is very time consuming to update._

### Use Case

_The content team/Info division would benefit from this added feature along with the regulated community._

### Misc

Here is a link to how the data was displayed in the 2018 election 
https://transition.fec.gov/info/charts_primary_dates_2018.shtml

A draft template is in Wagtail under " 2020 Dates and deadlines." It is a child page under this heading, which will display the lengthy html code when a chart is inserted as html code. 

### Future Work

_Other tables to be created in #3017_

### PR DESCRIPTION
**Pull request by johnnyporkchops**
## Summary (required)
This is the full-width page Wagtail template and accompanying JS that will allow us to achieve the mocked up experience and responsive design in issue #3016 
- Resolves #3016
_Include a summary of proposed changes._

## Impacted areas of the application
modified:   fec/static/scss/components/_table-styles.scss
modified:   home/models.py
modified:   fec/static/js/polyfills.js
modified:   ../package-lock.json
modified:   ../package.json

new: fec/static/js/pages/reporting-dates-tables.js
new: home/migrations/0105_auto_20190920_1216.py
new: home/templates/home/full_width_page.html

**To Do:** (All but the first one here are easy fixes)
- [ ] Resolve issue with using accordion on this page. Accordion does not work for the `All footnotes` section. `data-init.js` (required for the A11Y dialog) interferes with the accordion trigger. I have left the native `HTML5 Details` element here for now.
- [ ] Tweak some styling and make any other changes requested by @Jonella
- [ ] Clean up JS, review comments for brevity and clarity 
- [ ] Decide the cell-border color that works for both white and zebra-striped rows
- [ ] switch hex colors in SCSS with proper variables (ie: `#ddd` becomes `$gray-medium` or whatever)
- [ ]  SCSS - Remove as many `!important` rules as possible - some may not be necessary

## Screenshots
Screenshots to come soon

## How to test
Include any information that may be helpful to the reviewer(s).
1) Checkout `feature/3016-reporting-dates-table-template`
2) CD to /fec-cms and `rm -rf node_modules`
3)  `npm install`
4) `npm run build`
5) cd to fec-cms/fec/ and ` ./manage.py runserver`
6) go to `127.0.0.1/admin`
7) create a page of type `FullWidthPage`
8) create three html blocks and paste in the sections from the  attached  files into the three blocks. This step simulates the content team member copy/pasting from Excel but leaves out the actual spreadsheet part here.
[sample-data.txt](https://github.com/fecgov/fec-cms/files/3637508/sample-data.txt)

____

### CODE REVIEW

File: fec/home/models.py [hunk 1]
```diff
@@ -1201,3 +1201,18 @@ class ContactPage(Page):
     @property
     def content_section(self):
         return 'about'
+
+class FullWidthPage(ContentPage):
+    formatted_title = models.CharField(max_length=255, null=True, blank=True, default='',
```
Comment (patphongs) [inline fec/home/models.py]: @johnnyporkchops It appears that the page title is not inserting itself onto the page.

Screen Shot 2019-09-23 at 9 44 12 AM

Comment (johnnyporkchops) [reply fec/home/models.py]: @pat, I will push up a commit adding the title to the template. I have a few other IE fixes to push up as well.

File: fec/fec/static/js/pages/reporting-dates-tables.js [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+
+Array.from(date_sups)
+  .reverse()
+  .forEach(node => {
+    const indx = node.innerText;
+    const state_class = node.closest('tr').className;
+
+    const ftnt_row = `<tr class='${state_class} footnote_row footnote_${indx}'>
+                        <td></td>
+                        <td colspan='5'>
```
Comment (patphongs) [inline fec/fec/static/js/pages/reporting-dates-tables.js]: For future implementation with IEs, ECs, CCs, etc. There are different amounts of columns. IEs have 4 columns, the ECs and CCs both have 3 columns. Perhaps we should calculate colspan based on the count of `<th></th>` within a table?

Comment (johnnyporkchops) [reply fec/fec/static/js/pages/reporting-dates-tables.js]: I will make this change as well

File: fec/home/templates/home/full_width_page.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+{% block body_class %}template-{{ self.get_verbose_name | slugify }}{% endblock %}
+
+{% block content %}
+
+{% block breadcrumbs %}
+  {% include 'partials/breadcrumbs.html' with page=self links=self.get_ancestors style=self.breadcrumb_style %}
+{% endblock %}
+
+<article class="main container">
+	    <h2>{{ self.title }}</h2>
```
Comment (patphongs) [inline fec/home/templates/home/full_width_page.html]: I think we'll want the page title to be an H1? Like this:

`<h1 class="heading--main">{{ self.title }}</h1>`

File: fec/fec/static/js/pages/reporting-dates-tables.js [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+                        <td></td>
+                        <td colspan=${ftnt_colspan}>
+                          <b>${indx}.</b>&nbsp;${footnotes[indx]}</td>
+                      </tr>`;
+    node.closest('tr').insertAdjacentHTML('afterend', ftnt_row);
+  });
+
+//convert '**' to '*' in header notes for state (first td) cell
+//and wrap them in a link
+const state_cells = document.querySelectorAll('tr td:first-child');
```
Comment (patphongs) [inline fec/fec/static/js/pages/reporting-dates-tables.js]: There will be instances where `*` is used for footnotes other than the first column. Such as these FEA tables: https://transition.fec.gov/info/charts_fea_dates_2016.shtml. Can we change this to be looking in all cells?

Comment (johnnyporkchops) [reply fec/fec/static/js/pages/reporting-dates-tables.js]: @pat, I will put in a fix to handle “*” in footbotes

Comment (johnnyporkchops) [reply fec/fec/static/js/pages/reporting-dates-tables.js]: Also @djgarr said that the double asterisks ( `**`) on state name in the first column on the 2020 dates master spreadsheet are a typo and he is switching these all to single asterisks (`*`). We will have to add another footnote column for the state-name cells (the first column) so footnotes can be associated with those as well.

**Comment by johnnyporkchops** [comment]
Closing this PR in favor of #3227

**Pull request closed without merging**